app.rb

```diff
@@ -1,197 +1,116 @@
-    else
-      @people.each do |person|
-        puts "[#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
-      end
-    end
+  def list_people
+    @people.each { |person| puts person }
   end
 
-  # print out all people with numbers
-  def list_all_person_with_numbers
-    @people.each_with_index do |person, index|
-      puts "#{index}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+  def create_person
+    print 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
+    choice = gets.chomp
+    case choice
+    when '1'
+      create_student
+    when '2'
+      create_teacher
+    else
+      puts 'That is not a valid input'
+      nil
     end
   end
 
-  # create a student
   def create_student
     print 'Age: '
     age = gets.chomp
-
     print 'Name: '
     name = gets.chomp
-
-    print 'Has parent persmission? [Y/N]:'
-    permission = gets.chomp.downcase
-    case permission
-    when 'y' then parent_permission = true
-    when 'n' then parent_permission = false
-    end
-
-    # create a student and push it to people array
-    @people.push(Student.new('11', age, name, parent_permission: parent_permission))
+    print 'Has parent permission? [Y/N]: '
+    parent_permission = gets.chomp.downcase == 'y'
+    student = Student.new(age: age, name: name, parent_permission: parent_permission, classroom: @classroom)
+    @people.push(student)
+    puts 'Person created successfully'
   end
 
-  # create a teacher
   def create_teacher
     print 'Age: '
     age = gets.chomp
-
     print 'Name: '
     name = gets.chomp
-
     print 'Specialization: '
     specialization = gets.chomp
-
-    # create a teacher and push it to people array
-    @people.push(Teacher.new(specialization, age, name))
-  end
-
-  # create a person
-  def create_person
-    print 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
-    case gets.chomp
-    when '1'
-      create_student
-
-    when '2'
-      create_teacher
-    end
-
-    # success message
+    teacher = Teacher.new(age: age, name: name, specialization: specialization)
+    puts teacher.specialization
+    @people.push(teacher)
     puts 'Person created successfully'
   end
 
-  # create a book
   def create_book
     print 'Title: '
     title = gets.chomp
-
     print 'Author: '
     author = gets.chomp
-
-    # create the book object and add it to the books list
-    @books.push(Book.new(title, author))
-
-    # success message
+    book = Book.new(title, author)
+    @books.push(book)
     puts 'Book created successfully'
   end
 
-  # create a rental
-  def create_rental()
-    # select book
-    puts 'Select a book from the following list by number: '
-    list_all_books
-    book_option = gets.chomp.to_i
+  def create_rental
+    puts 'Select a book from the following list by number'
+    @books.each_with_index { |book, i| puts "#{i}) #{book}" }
```

Comment:
- [ ] When I select option 5 to create a rental and it lists the books the same issue is present here as well. You can change to this:

  ```
  @books.each_with_index { |book, i| puts "#{i}) #{book.title} #{book.author}" }
  ```